### PR TITLE
feat: support type-aware rules

### LINKED ISSUE #4: Support for type-aware rules
### Clear and concise description of the problem

Type-aware rules require type information to run. It would be great if type-aware rules "just worked".

### Suggested solution

Ideally, this should be all that is needed (when using `@typescript-eslint/parser` v8):

```ts
const { valid, invalid } = createRuleTester({
  name,
  rule,
  configs: {
    languageOptions: {
      parser: typescriptParser,
      parserOptions: {
        projectService: true, // or `project: true`
      },
    },
  },
});
```

I don't know what needs to be changed under the hood to make this actually work.

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by RebeccaStevens**
### Description

Adds support for testing type-aware rules.

### Linked Issues

fixes #4

### CODE REVIEW

**Review by pauliesnug — APPROVED** [review]

**Comment by RebeccaStevens** [comment]
@antfu ping

File: src/rule-tester.ts [hunk 1]
```diff
@@ -23,8 +38,16 @@ export function createRuleTester(options: RuleTesterInitOptions): RuleTester {
       defaultConfigs.unshift(inlineConfig)
   }
 
+  const defaultFilenames = {
+    js: 'file.js',
+    ts: 'file.ts',
+    jsx: 'react.jsx',
+    tsx: 'react.tsx',
```
Comment (antfu) [inline src/rule-tester.ts:45]:
```suggestion
    jsx: 'file.jsx',
    tsx: 'file.tsx',
```

Wouldn't it better just having `file` and be more natural? Any reason `react` is needed?

**Review by antfu — APPROVED** [review]
LGTM, thanks

File: src/rule-tester.ts (hunk 1 above)
Comment (RebeccaStevens) [reply src/rule-tester.ts]: That's just how typescript-eslint does it for their tester so I just copied them.

**Comment by pauliesnug** [comment]
@antfu Is there any docs for this change? If not that's fine

**Pull request merged**